warehouse/oidc/models/_core.py

```diff
@@ -136,7 +137,7 @@ def lookup_by_claims(cls, session, signed_claims: SignedClaims):
 
             if publisher := query.with_session(session).one_or_none():
                 return publisher
-        return None
+        raise InvalidPublisherError("All lookup strategies exhausted")
```

Comment:
Nice! This will be especially handy in terms of distinguishing logic errors in lookup vs. claim verification.